GVFS/GVFS.Common/Maintenance/LooseObjectsStep.cs

```diff
@@ -22,26 +23,63 @@ public LooseObjectsStep(GVFSContext context, bool requireCacheLock, bool forceRu
 
         public override string Area => nameof(LooseObjectsStep);
 
+        public virtual int MaxLooseObjectsInPack => 1000;
+
         protected override string LastRunTimeFilePath => Path.Combine(this.Context.Enlistment.GitObjectsRoot, "info", LooseObjectsLastRunFileName);
         protected override TimeSpan TimeBetweenRuns => TimeSpan.FromDays(7);
 
         public int CountLooseObjects()
         {
             int count = 0;
-            
+
             foreach (string directoryPath in this.Context.FileSystem.EnumerateDirectories(this.Context.Enlistment.GitObjectsRoot))
             {
                 string directoryName = directoryPath.TrimEnd(Path.DirectorySeparatorChar).Split(Path.DirectorySeparatorChar).Last();
 
                 if (GitObjects.IsLooseObjectsDirectory(directoryName))
-                {                    
+                {
                     count += this.Context.FileSystem.GetFiles(Path.Combine(this.Context.Enlistment.GitObjectsRoot, directoryPath), "*").Count();
                 }
             }
 
             return count;
         }
 
+        public void PackLooseObjects(StreamWriter streamWriter)
+        {
+            // Find looseObjects
+            foreach (string directoryPath in this.Context.FileSystem.EnumerateDirectories(this.Context.Enlistment.GitObjectsRoot))
+            {
+                string directoryName = directoryPath.TrimEnd(Path.DirectorySeparatorChar).Split(Path.DirectorySeparatorChar).Last();
+
+                if (GitObjects.IsLooseObjectsDirectory(directoryName))
+                {
+                    string[] looseObjectFileNamesInDir = this.Context.FileSystem.GetFiles(Path.Combine(this.Context.Enlistment.GitObjectsRoot, directoryPath), "*");
+
+                    foreach (string file in looseObjectFileNamesInDir)
+                    {
+                        if (this.LooseObjectsPutIntoPackFile == this.MaxLooseObjectsInPack)

```

Comment:
Maybe test >= rather than exact equality, in case we decide one day we'll add in batches.